### PR TITLE
more mobile send: send from profile

### PR DESCRIPTION
**Pull request by chrisnojima**
this enables send from profile and fixes some random things i saw

### CODE REVIEW

File: shared/people/container.js [hunk 1]
```diff
@@ -28,7 +27,7 @@ const mapStateToProps = state => ({
   waiting: WaitingConstants.anyWaiting(state, getPeopleDataWaitingKey),
 })
 
-const mapDispatchToProps = dispatch => ({
+const mapDispatchToProps = (dispatch, {navigateAppend}) => ({
```
Comment (chrisnojima) [inline shared/people/container.js]: if you click search too quickly you can get a redbox

File: shared/actions/wallets.js [hunk 2]
```diff
@@ -235,6 +235,7 @@ const loadSendAssetChoices = (state: TypedState, action: WalletsGen.LoadSendAsse
   })
 
 const loadDisplayCurrency = (state: TypedState, action: WalletsGen.LoadDisplayCurrencyPayload) =>
+  action.payload.accountID !== Types.noAccountID &&
```
Comment (chrisnojima) [inline shared/actions/wallets.js]: we can call this with noAccountID and get a black bar

File: shared/react-native/ios/Keybase/NativeLogger.m [hunk 3]
```diff
@@ -27,20 +27,22 @@ + (BOOL)requiresMainQueueSetup
...
-  AppDelegate *appDel = (AppDelegate *)[[UIApplication sharedApplication] delegate];
-  DDFileLogger *fileLogger = (DDFileLogger *)appDel.fileLogger;
-  NSArray<NSString *> *paths = [[fileLogger logFileManager] sortedLogFilePaths];
-  for (NSString *path in paths) {
-    NSString *fileContents = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:NULL];
-    for (NSString *line in [fileContents componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]]) {
-      NSRange range = [line rangeOfString:[NSString stringWithFormat:@"%@%@: ", tagPrefix, tagName]];
-      if (range.location != NSNotFound) {
-        [lines addObject:[line substringFromIndex:range.location + range.length]];
+  dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/NativeLogger.m]: this likely needs to only happen in the main queue. access to the AppDelegate for sure but likely this is all bad to touch from multiple threads

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: shared/react-native/ios/Keybase/NativeLogger.m (hunk 3 above)
Comment (mmaxim) [reply shared/react-native/ios/Keybase/NativeLogger.m]: Maybe better for this to have its own queue?

File: shared/actions/wallets.js (hunk 2 above)
Comment (zanderz) [reply shared/actions/wallets.js]: would isValidAccountID be better?

Comment (chrisnojima) [reply shared/actions/wallets.js]: unclear, there's just one value so it doesn't seem necessary

File: shared/react-native/ios/Keybase/NativeLogger.m (hunk 3 above)
Comment (chrisnojima) [reply shared/react-native/ios/Keybase/NativeLogger.m]: @mmaxim done

**Comment by chrisnojima** [comment]
you aren't supposed to touch app delegate from a non-ui thread, so we bounce to that thread to get it, then bounce to a background thread to do the log dump

**Pull request merged**